proposals/improved-interpolated-strings.md

```diff
@@ -24,47 +24,41 @@ repeatedly call `AppendFormat` with every part, and then get a final string out.
-that a user writes. It also extends this pattern to allow for interpolated strings passed as arguments to other methods to use a builder pattern, defined by
+that a user writes. It also extends this pattern to allow for interpolated strings passed as arguments to other methods to use a handler pattern, defined by
 receiver of the method, that will allow things like logging frameworks to avoid allocating strings that will never be needed, and giving C# users familiar,
 convenient interpolation syntax.
 
 ## Detailed Design
 
-### The builder pattern
+### The handler pattern
 
-We introduce a new builder pattern that can represent an interpolated string passed as an argument to a method. The simple English of the pattern is as follows:
+We introduce a new handler pattern that can represent an interpolated string passed as an argument to a method. The simple English of the pattern is as follows:
 
-When an _interpolated\_string\_expression_ is passed as an argument to a method, we look at the type of the parameter. If the parameter type has a static method
-`Create` that can be invoked with 2 int parameters, `literalLength` and `formattedCount`, optionally takes a parameter the receiver is convertible to,
+When an _interpolated\_string\_expression_ is passed as an argument to a method, we look at the type of the parameter. If the parameter type has a constructor
+that can be invoked with 2 int parameters, `literalLength` and `formattedCount`, optionally takes a parameter the receiver is convertible to,
 and has an out parameter of the type of original method's parameter and that type has instance `AppendLiteral` and `AppendFormatted` methods that
 can be invoked for every part of the interpolated string, then we lower the interpolation using that, instead of into a traditional call to
 `string.Format(formatStr, args)`. A more concrete example is helpful for picturing this:
 
 ```cs
-// The builder that will actually "build" the interpolated string"
-[InterpolatedStringBuilder]
-public ref struct TraceLoggerParamsBuilder
+// The handler that will actually "build" the interpolated string"
+[InterpolatedStringHandler]
+public ref struct TraceLoggerParamsInterpolatedStringHandler
 {
-    public static TraceLoggerParamsBuilder Create(int literalLength, int formattedCount, Logger logger, out bool builderIsValid)
-    {
-        if (!logger._logLevelEnabled)
-        {
-            builderIsValid = false;
-            return default;
-        }
-
-        builderIsValid = true;
-        return TraceLoggerParamsBuilder(literalLength, formattedCount, logger.EnabledLevel);
-    }
-
     // Storage for the built-up string
 
     private bool _logLevelEnabled;
 
-    private TraceLoggerParamsBuilder(int literalLength, int formattedCount, bool logLevelEnabled)
+    public TraceLoggerParamsInterpolatedStringHandler(int literalLength, int formattedCount, Logger logger, out bool handlerIsValid)
```

Comment:
Just to confirm, the name of the out bool at the end isn't important, just its position, yes?

Reply:
That's correct.